pkg/skaffold/deploy/docker/deploy.go

```diff
@@ -138,23 +151,68 @@ func (d *Deployer) deploy(ctx context.Context, out io.Writer, b graph.Artifact)
 		return err
 	}
 
-	containerName := d.getContainerName(ctx, b.ImageName)
+	containerCfg, err := d.containerConfigFromImage(ctx, b.Tag)
+	if err != nil {
+		return err
+	}
+	containerCfg.ExposedPorts = ports
+
+	initContainers, err := d.debugger.TransformImage(ctx, b, containerCfg)
+	if err != nil {
+		return errors.Wrap(err, "transforming image for debugging")
+	}
+
+	for _, c := range initContainers {
+		if d.debugger.HasMount(c.Image) {
```

Comment:
It feels a bit odd to have this code run all the initContainers but the debugManager record the details. I guess that allows sharing the initContainers across deployers. But that also assumes that we'll use the same docker daemon? That's worth noting here in case we ever support multiple docker daemons.

Reply:
that's a great point actually, this code never assumes anything but the docker daemon configured implicitly on the system. i'll put a note here as a bread crumb in case we ever go that route.

the main reason we need this loop is because across subsequent deploys, we could have new transforms applied leading to new init containers, but the transformer itself doesn't know which have already been applied, so the `DebugManager` tracks it.